### PR TITLE
fix to failing view.exportAsImage

### PR DESCRIPTION
**Pull request by ksobon**
### Purpose

This addresses this issue: https://github.com/DynamoDS/Dynamo/issues/3262 (possibly but it's an old one) and this issue: https://github.com/DynamoDS/DynamoRevit/issues/1695

Basically the idea is that we use ImageExportOptions.GetFileName instead of manually constructing the predicted filename that Revit will assign to our export. It fails. 

image

Using their built in static method does not:

image

### Declarations

Check these if you believe they are true

- [x] The code base is in a better state after this PR
- [x] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [x] The level of testing this PR includes is appropriate
- [x] User facing strings, if any, are extracted into `*.resx` files
- [x] Snapshot of UI changes, if any.

### Reviewers

@mjkkirschner

### CODE REVIEW

File: src/Libraries/RevitNodes/Elements/Views/View.cs [hunk 1]
```diff
@@ -99,25 +100,34 @@ public Bitmap ExportAsImage(string path)
...
 
-            // and the intended destination
-            var destFn = pathName + extension;
+            if(string.IsNullOrEmpty(actualFileName) || string.IsNullOrEmpty(destinationFileName))
+                throw new ArgumentException(Properties.Resources.View_ExportAsImage_Path_Invalid, nameof(path));
 
             // rename the file
-            if (File.Exists(destFn)) File.Delete(destFn);
-            File.Move(actualFn, destFn);
+            if (File.Exists(destinationFileName)) File.Delete(destinationFileName);
```
Comment (dimven) [inline src/Libraries/RevitNodes/Elements/Views/View.cs]: What happens if the user has the old image file open and there's a file lock in place? You might want to wrap this in a try/catch or check for a file lock

Comment (ksobon) [inline src/Libraries/RevitNodes/Elements/Views/View.cs]: I don't remember seeing anything that would handle that in DynamoRevit source code. I can certainly look at that.

**Comment by Racel** [comment]
Added to our backlog. Will run RTF and get back to you. We will handle the merge conflicts.

**Comment by Racel** [comment]
@ksobon - We are trying to go through several of your PRs. It appears as though this PR has merge conflicts. Can you please let us know if you will have time to fix these issues in the next couple of weeks? If not, please close and reopen at a later date.

**Comment by Racel** [comment]
@ksobon - As per our discussion, we are closing this task. Please feel free to reopen at a later date.

**Pull request closed without merging**